rdflib/namespace/__init__.py

```diff
@@ -590,6 +590,31 @@ def compute_qname_strict(
 
             return self.__cache_strict[uri]
 
+    def expand_curie(self, curie: str) -> Union[URIRef, None]:
+        """
+        Expand a qname (aka CURIE) of the form <prefix:element>, e.g. "rdf:type"
+        into its full expression:
+
+        >>> import rdflib
+        >>> g = rdflib.Graph()
+        >>> g.namespace_manager.expand_curie("rdf:type")
+        'http://www.w3.org/1999/02/22-rdf-syntax-ns#type'
+
+        Returns None if a namespace is not bound to the prefix.
+
+        """
+        if not type(curie) is str:
+            raise TypeError("Argument must be a string.")
+        elif ":" not in curie or curie.split(":")[0] == "" or curie.split(":")[1] == "":
+            raise ValueError(
+                "Malformed curie argument, format should be e.g. “foaf:name”."
+            )
+        else:
+            ns = self.store.namespace(curie.split(":")[0])
+            if ns is not None:
+                return URIRef(str(ns) + curie.split(":")[1])
+        return None
```

Comment:
I'm also a bit on the fence on returning None, I would just raise an error, but I would not say this is wrong, would like to hear some other opinions on the matter.

CC: @RDFLib/core-reviewers

Reply:
I prefer raising an error: if the thig ased for cannot be calculated, an error should be returned, not None which just indicates it doesn't exist.